---
title: "SynergyFinder Report"
author: "SynergyFinder (http://synergyfinder.org/)"
date: "`r Sys.Date()`"
header-includes:
 - \usepackage{booktabs}
 - \usepackage{longtable}
 - \usepackage{array}
 - \usepackage{multirow}
 - \usepackage{wrapfig}
 - \usepackage{float}
 - \usepackage{colortbl}
 - \usepackage{pdflscape}
 - \usepackage{tabu}
 - \usepackage{threeparttable}
 - \usepackage{threeparttablex}
 - \usepackage[normalem]{ulem}
 - \usepackage{makecell}
 - \usepackage{xcolor}
output: 
  pdf_document: default
fig_caption: yes
params:
  data: NA
  blocks: NA
  correct_baseline: NA
  DRC_plots: NA
  DR_multi_high_value_color: NA
  DR_multi_low_value_color: NA
  DR_multi_point: NA
  DR_multi_point_color: NA
  DR_plot_type: NA
  DR_rep_statistic: NA
  DR_summary_statistic: NA
  DR_high_value_color: NA
  DR_low_value_color: NA
  DR_heatmap_label_color: NA
  DR_grid: NA
  syn_multi_high_value_color: NA
  syn_multi_low_value_color: NA
  syn_multi_point: NA
  syn_multi_point_color: NA
  syn_plot_type: NA
  syn_summary_statistic: NA
  syn_high_value_color: NA
  syn_low_value_color: NA
  syn_heatmap_label_color: NA
  syn_heatmap_label_size: NA
  syn_text_size: NA
  syn_rep_statistic: NA
  syn_grid: NA
  bb_panel_title_size: NA
  bb_axis_text_size: NA
  bb_highlight_label_size: NA
  bb_highlight_pos_color: NA
  bb_highlight_neg_color: NA
  bb_pos_value_color: NA
  bb_neg_value_color: NA
  ss_point_color: NA
  ss_point_size: NA
  ss_show_label: NA
  ss_label_color: NA
  ss_label_size: NA
always_allow_html: true
---

```{r setup, include=FALSE}
options(kableExtra.auto_format= FALSE)
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, error = FALSE, fig.pos = "H")
options(tinytex.verbose = TRUE)
```
```{r include=FALSE}
library(tidyverse)
library(synergyfinder)
library(kableExtra)
library(synergyfinder)
synergyfinder_obj <- params$data
ndrug <- sum(grepl("drug", colnames(synergyfinder_obj$drug_pairs)))
methods = c("ZIP", "Loewe", "Bliss", "HSA")

if (is.null(params$DR_multi_high_value_color)) {
  DR_multi_high_value_color <- "#CC3311"
} else {
  DR_multi_high_value_color <- params$DR_multi_high_value_color
}
if (is.null(params$DR_multi_low_value_color)) {
  DR_multi_low_value_color <- "#448BD4"
} else {
  DR_multi_low_value_color <- params$DR_multi_low_value_color
}
if (is.null(params$DR_multi_point_color)) {
  DR_multi_point_color <- "#DDA137"
} else {
  DR_multi_point_color <- params$DR_multi_point_color
}
if (is.null(params$DR_multi_point)) {
  DR_multi_point <- TRUE
} else {
  DR_multi_point <- params$DR_multi_point
}
if (is.null(params$syn_multi_high_value_color)) {
  syn_multi_high_value_color <- "#CC3311"
} else {
  syn_multi_high_value_color <- params$syn_multi_high_value_color
}
if (is.null(params$syn_multi_low_value_color)) {
  syn_multi_low_value_color <- "#448BD4"
} else {
  syn_multi_low_value_color <- params$syn_multi_low_value_color
}
if (is.null(params$syn_multi_point_color)) {
  syn_multi_point_color <- "#DDA137"
} else {
  syn_multi_point_color <- params$syn_multi_point_color
}
if (is.null(params$syn_multi_point)) {
  syn_multi_point <- TRUE
} else {
  syn_multi_point <- params$syn_multi_point
}
```

```{r calculate_setting}
cat("Correct Baseline Setting: ", params$correct_baseline)
```

# 1. Data tables

```{r meta_table}
meta_table <- synergyfinder_obj$drug_pairs %>% 
  dplyr::filter(block_id == params$blocks) %>% 
  dplyr::select(
    "Block ID" = "block_id",
    dplyr::starts_with("drug"),
    dplyr::starts_with("conc_unit"),
    dplyr::starts_with("cell"))
names(meta_table)[startsWith(names(meta_table), "conc_unit")] <- paste0(
    gsub(
      "conc_unit",
      "Conc Unit\\\\textsubscript{",
     names(meta_table)[startsWith(names(meta_table), "conc_unit")]
    ),
    "}"
  )
names(meta_table)[startsWith(names(meta_table), "drug")] <- paste0(
  gsub(
    "drug",
    "Drug\\\\textsubscript{",
    names(meta_table)[startsWith(names(meta_table), "drug")]
  ),
  "}"
)
names(meta_table)[names(meta_table) == "cell_line_name"] <- "Cell Line Name"
meta_table %>%
  mutate(across(where(is.numeric), round, 2)) %>% 
  kableExtra::kable(
    "latex",
    booktabs = TRUE,
    caption = "Drug Combiantion Meta Data",
    escape = FALSE
  ) %>%
  kable_styling(latex_options = c("striped", "HOLD_position"))
```

```{r synergy_table}
df_synergy <- synergyfinder_obj$drug_pairs %>% 
  dplyr::filter(block_id == params$blocks) %>% 
  dplyr::select(
    "Block ID" = "block_id",
    "ZIP" = "ZIP_synergy",
    "Loewe" = "Loewe_synergy",
    "HSA" = "HSA_synergy",
    "Bliss" = "Bliss_synergy"
    )
names(df_synergy) <- gsub("_", " ", names(df_synergy))
df_synergy %>% 
  mutate(across(where(is.numeric), round, 2)) %>% 
  kableExtra::kable(
    "latex",
    caption = "Synergy Score Summary Table",
    escape = FALSE) %>%
  kableExtra::kable_styling(latex_options = c("striped", "HOLD_position"))
```

```{r synergy_statistic_table}
if (any(synergyfinder_obj$drug_pairs$replicate)) {
  df_synergy_statistic <- synergyfinder_obj$drug_pairs %>% 
  dplyr::filter(block_id == params$blocks) %>% 
  dplyr::select(
    "Block ID" = "block_id",
    `ZIP p Value` = "ZIP_synergy_p_value",
    `Loewe p Value` = "Loewe_synergy_p_value",
    `HSA p Value` = "HSA_synergy_p_value",
    `Bliss p Value` = "Bliss_synergy_p_value"
    )
names(df_synergy_statistic) <- gsub("_", " ", names(df_synergy_statistic))
df_synergy_statistic %>%
  mutate(across(where(is.numeric), round, 2)) %>% 
  kableExtra::kable(
    "latex",
    caption = "Synergy Score Statistic Table",
    escape = FALSE) %>%
  kableExtra::kable_styling(latex_options = c("striped", "HOLD_position"))
}
```

```{r sensitivity_table}
df_sensitive <- synergyfinder_obj$drug_pairs %>% 
  dplyr::filter(block_id == params$blocks) %>% 
  dplyr::select(
    "Block ID" = "block_id",
    dplyr::contains("ri_"),
    "CSS" = "css")
names(df_sensitive)[startsWith(names(df_sensitive), "ri_")] <- paste0(
  gsub(
    "ri_",
    "RI\\\\textsubscript{",
    names(df_sensitive)[startsWith(names(df_sensitive), "ri_")]
  ),
  "}"
)

df_sensitive %>% 
  mutate(across(where(is.numeric), round, 2)) %>% 
  kableExtra::kable(
    "latex",
    caption = "Sensitivity Score Summary Table",
    escape = FALSE
  ) %>%
  kableExtra::kable_styling(latex_options = c("striped", "HOLD_position"))
```

```{r sensitivity_statistic_table}
if (any(synergyfinder_obj$drug_pairs$replicate)) {
  df_sensitive_statistic <- synergyfinder_obj$sensitivity_scores_statistics %>% 
    dplyr::filter(block_id == params$blocks) %>% 
    dplyr::select(
    "Block ID" = "block_id",
      paste0("ri_", 1:ndrug, "_p_value"),
      "CSS p Value" = "css_p_value"
      ) 
  names(df_sensitive_statistic)[startsWith(names(df_sensitive_statistic), "ri_")] <-
    gsub(
      "_p_value",
      "} p Value",
      gsub(
        "ri_",
        "RI\\\\textsubscript{",
        names(df_sensitive_statistic)[startsWith(names(df_sensitive_statistic), "ri_")]
     )
    )
  df_sensitive_statistic %>% 
  mutate(across(where(is.numeric), round, 2)) %>% 
  # knitr::kable(caption = "Synergy Score Statistic Table", format = "latex")
  kableExtra::kable(
    "latex",
    caption = "Sensitivity Score Statistic Table",
    escape = FALSE) %>%
  # kableExtra::kable_classic(full_width = F) %>%
  kableExtra::kable_styling(latex_options = c("HOLD_position"))
}
```

# 2. Dose response curve

```{r load_dose_response_curve, fig.show="hold", out.width="50%"}
knitr::include_graphics(params$DRC_plots)
```

# 3. Dose response map

```{r dose_response_map_heatmap, fig.show="hold", out.width="100%"}
if (params$DR_plot_type == "heatmap"){
  for (b in params$blocks) {
    comb <- combn(1:ndrug, 2)
    for (c in 1:ncol(comb)) {
      p <- Plot2DrugHeatmap(
        data = synergyfinder_obj,
        plot_block = b,
        plot_title = paste0("Block ", b, " Dose Response Matrix"),
        drugs = c(comb[1, c], comb[2, c]),
        dynamic = FALSE,
        statistic = params$DR_rep_statistic,
        summary_statistic = params$DR_summary_statistic,
        high_value_color = params$DR_high_value_color,
        low_value_color = params$DR_low_value_color,
        # text_label_size_scale = params$DR_heatmap_label_size,
        text_label_color = params$DR_heatmap_label_color
        # title_text_size_scale = params$DR_text_size
      )
      print(p)
    }
  }
}

```

```{r dose_response_map, fig.show="hold", out.width="50%"}
if (params$DR_plot_type != "heatmap") {
  for (b in params$blocks) {
    comb <- combn(1:ndrug, 2)
    for (c in 1:ncol(comb)) {
      p <- Plot2DrugSurface(
        data = synergyfinder_obj,
        drugs = c(comb[1, c], comb[2, c]),
        plot_block = b,
        plot_title = paste0("Block ", b, " Dose Response Matrix"),
        dynamic = FALSE,
        summary_statistic = params$DR_summary_statistic,
        high_value_color = params$DR_high_value_color,
        low_value_color = params$DR_low_value_color,
        text_size_scale = 1,
        grid = params$DR_grid
        # text_size_scale = params$DR_text_size
      )
      print(p)
    }
  }
}

```

# 4. Synergy Scores

```{r synergy_plots_heatmap, fig.show="hold", out.width="100%"}
if (params$syn_plot_type == "heatmap") {
  for (b in params$blocks) {
    comb <- combn(1:ndrug, 2)
    # 2-drug synergy surface
    for (c in 1:ncol(comb)) {
      for (m in methods) {
      p <- PlotSynergy(
        data = synergyfinder_obj,
        block_ids = b,
        plot_title = paste("Block", b, m, "Synergy Score", sep = " "),
        drugs = c(comb[1, c], comb[2, c]),
        type = params$syn_plot_type,
        method = m,
        dynamic = FALSE,
        summary_statistic = params$syn_summary_statistic,
        high_value_color = params$syn_high_value_color,
        low_value_color = params$syn_low_value_color,
        text_size_scale = params$syn_text_size,
        heatmap_text_label_color = params$syn_heatmap_label_color,
        heatmap_text_label_size_scale = params$syn_heatmap_label_size,
        statistic = params$syn_rep_statistic,
        grid = params$syn_grid,
        # text_size_scale = 1.5,
        display = FALSE
      )
      print(p[[1]])
      }
    }
  }
}
```

```{r, synergy_plots, fig.show="hold", out.width="50%"}
if (params$syn_plot_type != "heatmap") {
  for (b in params$blocks) {
    comb <- combn(1:ndrug, 2)
    # 2-drug synergy surface
    for (c in 1:ncol(comb)) {
      for (m in methods) {
      p <- PlotSynergy(
        data = synergyfinder_obj,
        block_ids = b,
        plot_title = paste("Block", b, m, "Synergy Score", sep = " "),
        drugs = c(comb[1, c], comb[2, c]),
        type = params$syn_plot_type,
        method = m,
        dynamic = FALSE,
        summary_statistic = params$syn_summary_statistic,
        high_value_color = params$syn_high_value_color,
        low_value_color = params$syn_low_value_color,
        text_size_scale = params$syn_text_size,
        heatmap_text_label_color = params$syn_heatmap_label_color,
        heatmap_text_label_size_scale = params$syn_heatmap_label_size,
        statistic = params$syn_rep_statistic,
        grid = params$syn_grid,
        # text_size_scale = 1.5,
        display = FALSE
      )
      print(p[[1]])
      }
    }
  }
}
```

# 5. Bar plots

```{r results='asis'}
  # multi-drug bar plot
for (b in params$blocks) {
  p <- PlotMultiDrugBar(
    data = synergyfinder_obj,
    plot_block = b,
    plot_value = c(
      "response", "ZIP_synergy", "HSA_synergy",
      "Bliss_synergy", "Loewe_synergy"
    ),
    highlight_row = NULL,
    sort_by = "response",
    panel_title_size = params$bb_panel_title_size * 0.6,
    axis_text_size = params$bb_axis_text_size * 0.6,
    highlight_label_size = params$bb_highlight_label_size * 0.6,
    highlight_pos_color = params$bb_highlight_pos_color,
    highlight_neg_color = params$bb_highlight_neg_color,
    pos_value_color = params$bb_pos_value_color,
    neg_value_color = params$bb_neg_value_color,
    data_table = FALSE
  )
  print(p)
}
```

# 6. S-S plot

```{r ss_plot, fig.show="hold", out.width="100%"}
for (m in methods) {
  p <- PlotSensitivitySynergy(
    data = synergyfinder_obj,
    plot_synergy = m,
    plot_title = paste(m, "- CSS", sep = " "),
    point_color = params$ss_point_color,
    point_size = 3,
    show_labels = params$ss_show_label,
    point_label_color = params$ss_label_color,
    label_size = 14,
    text_size_scale = 1.5,
    dynamic = FALSE
  )
  print(p)
}
```

# Reference

### For use of SynergyFinder R package or web application:

Zheng, S.; Wang, W.; Aldahdooh, J.; Malyutina, A.; Shadbahr, T.; Tanoli, Z.; Passia, A.; Tang, J. SynergyFinder Plus: Toward Better Interpretation and Annotation of Drug Combination Screening Datasets. Genomics, Proteomics & Bioinformatics 2022. https://doi.org/10.1016/j.gpb.2022.01.004.

### For use of ZIP synergy scoring:

Yadav, B.; Wennerberg, K.; Aittokallio, T.; Tang, J. Searching for Drug Synergy in Complex Dose-Response Landscapes Using an Interaction Potency Model. Comput Struct Biotechnol J 2015, 13, 504–513. doi:10.1016/j.csbj.2015.09.001

### For how to harmonize the different synergy scoring methods:

Tang, J.; Wennerberg, K.; Aittokallio, T. What Is Synergy? The Saariselkä Agreement Revisited. Front Pharmacol 2015, 6, 181. doi:10.3389/fphar.2015.00181

### For general ideas of drug combination therapies:

Tang, J. Informatics Approaches for Predicting, Understanding, and Testing Cancer Drug Combinations. Methods Mol Biol 2017, 1636, 485–506. doi:10.1007/978-1-4939-7154-1_30

### For retrieving the most comprehensive drug combination data resources and their sensitivity and synergy results by SynergyFinder, please go to DrugComb :

Zheng, S.; Aldahdooh, J.; Shadbahr, T.; Wang, Y.; Aldahdooh, D.; Bao, J.; Wang, W.; Jing, T. DrugComb update: a more comprehensive drug sensitivity data repository and analysis portal. Nucleic Acids Research (2021) doi:10.1093/nar/gkab438

Zagidullin, B.; Aldahdooh, J.; Zheng, S.; Wang, W.; Wang, Y.; Saad, J.; Malyutina, A.; Jafari, M.; Tanoli, Z.; Pessia, A.; Tang, J. DrugComb: An Integrative Cancer Drug Combination Data Portal. Nucleic Acids Res 2019, 47 (W1), W43–W51. doi:10.1093/nar/gkz337

### For use of combination sensitivity score:

Malyutina, A.; Majumder, M. M.; Wang, W.; Pessia, A.; Heckman, C. A.; Tang, J. Drug Combination Sensitivity Scoring Facilitates the Discovery of Synergistic and Efficacious Drug Combinations in Cancer. PLOS Computational Biology 2019, 15 (5), e1006752. doi:10.1371/journal.pcbi.1006752